### ISSUE TITLE
attr-validate can cause interpreter to lock up

### ISSUE THREAD

**Issue by outoftime** [htmllint/htmllint#209]
Running `htmllint` with defaults on this HTML document:

```html
<!DOCTYPE html>
<html>
    <body>
        <a href=ddddddddddddddddddddddd">Hi</a>
    </body>
</html>
```

The JS interpreter will lock up for a substantial period of time.

Making the bogus URL longer causes the lockup time to increase precipitously.

I used `git-bisect` to track it down to https://github.com/htmllint/htmllint/commits/9607be9360aa152c7d95c1e2d820d4d230c220cf

I can confirm that disabling the `attr-validate` rule removes the lockup.

The general behavior here is consistent with [catastrophic backtracking](https://www.regular-expressions.info/catastrophic.html), though I haven’t taken a close look at the regular expression used for this particular rule.

**Comment by yvanavermaet** [comment]
I can't simulate the issue. Do you have a full testcase how I can experience your problem?

Also, seems there is an issue with following semver, unless we're going with the story of "this is still a beta, because we're still on "0.x.x and therefor we are allowed to do breaking changes"? @mlochbaum

**Comment by mlochbaum** [comment]
The problem was that the regex used in attr-validate.js doesn't require spaces between the matched "attributes", so for the given invalid string it tries starting new attributes in the middle of the URL to see if those work. Since it's possible (likely, in fact), that there will be no spaces on either side of the attribute string, changing `\s*` to `\s+` will break things. I decided to just append a space before checking rather than write a regex with two copies of the subexpression.
